cwl/src/main/scala/cwl/CwlFile.scala

```diff
@@ -71,12 +72,8 @@ case class CommandLineTool private(
   }
 
   object ArgumentToId extends Poly1 {
-    implicit def ecmaScript = at[ECMAScript] {
-      _.value
-    }
-
-    implicit def ecmaFunction = at[ECMAFunction] {
-      _.value
+    implicit def ecmaScript:Case.Aux[Expression, String] = at[Expression] {
+      _ => ???
```

Comment:
I think we came up with a good way to generate an ID for a `CommandLineTool => TaskDefinition`, but it seems it has not been implemented yet.  In any event I don't think an expression is a good candidate for an ID.  I'll create an issue if none has been filed.